Sources/StreamChat/Config/ChatClientConfig.swift

```diff
@@ -171,3 +176,22 @@ public struct APIKey: Equatable {
         self.apiKeyString = apiKeyString
     }
 }
+
+public extension ChatClientConfig {
+    /// Allows to inject a custom API client for uploading attachments, if not specified `StreamCDNClient` is used
+    @available(*, deprecated, message: "Set `cdnClientConfig = .custom(client)` instead")
+    var customCDNClient: CDNClient? {
+        get {
+            guard case let .custom(client) = cdnClientConfig else { return nil }
+            
+            return client
+        }
+        set {
+            if let customClient = newValue {
+                cdnClientConfig = .custom(client: customClient)
+            } else {
+                cdnClientConfig = .default(type: StreamCDNClient.self)
+            }
+        }
+    }
```

Comment:
No need to add deprecations, V4 will include breaking changes, so we can break the API for now 👍 Anyways, all deprecations should live in the `Deprecations.swift` file.